File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/movie/model/MovieCollection.java
package dev.reuss.tmdb.domain.movie.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Collection a movie belongs to.
 *
 * @param id           TMDB collection id
 * @param name         collection name
 * @param posterPath   poster image path
 * @param backdropPath backdrop image path
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record MovieCollection(
        int id,

        String name,

        @JsonProperty("poster_path")
        String posterPath,

        @JsonProperty("backdrop_path")
        String backdropPath
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/movie/model/MovieCollection.kt
package dev.reuss.tmdb.domain.movie.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Collection a movie belongs to.
 *
 * @property id TMDB collection id
 * @property name collection name
 * @property posterPath poster image path
 * @property backdropPath backdrop image path
 */
@JvmRecord
data class MovieCollection(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("name")
    val name: String?,
    @all:JsonProperty("poster_path")
    val posterPath: String?,
    @all:JsonProperty("backdrop_path")
    val backdropPath: String?,
) : TmdbModel
